service/src/main/java/com/codeforcommunity/processor/ProtectedUserProcessorImpl.java

```diff
@@ -95,4 +95,41 @@ public void changeEmail(JWTData userData, ChangeEmailRequest changeEmailRequest)
         AuthDatabaseOperations.getFullName(user.into(Users.class)),
         changeEmailRequest.getNewEmail());
   }
+
+  @Override
+  public void changePrivilegeLevel(
+      JWTData userData, ChangePrivilegeLevelRequest changePrivilegeLevelRequest) {
+    UsersRecord user =
+        db.selectFrom(USERS)
+            .where(USERS.EMAIL.eq(changePrivilegeLevelRequest.getTargetUserEmail()))
+            .fetchOne();
+    if (user == null) {
+      throw new UserDoesNotExistException(changePrivilegeLevelRequest.getTargetUserEmail());
+    }
+
+    // check if user is admin
+    if (!(userData.getPrivilegeLevel() == PrivilegeLevel.ADMIN
```

Comment:
This check should always come first in the method. Otherwise we're letting non-admin users check any email they want to see if it exists in our db